app/src/main/java/com/exam/myapp/MyApp.kt

```diff
@@ -0,0 +1,18 @@
+package com.exam.myapp
+
+import android.content.Context
+import com.exam.myapp.di.component.DaggerAppComponent
+import dagger.android.AndroidInjector
+import dagger.android.DaggerApplication
+
+open class MyApp : DaggerApplication() {
+
+    private lateinit var androidInjector: AndroidInjector<out DaggerApplication>
+
+    override fun applicationInjector(): AndroidInjector<out DaggerApplication> = androidInjector
+
+    override fun attachBaseContext(base: Context?) {
+        super.attachBaseContext(base)
+        androidInjector = DaggerAppComponent.builder().application(this).build()
+    }
```

Comment:
Dagger is good but we can use HILT as it's more improved and less code 

https://developer.android.com/training/dependency-injection/hilt-android